module/zfs/dnode.c

```diff
@@ -1441,11 +1441,7 @@ dnode_setdirty(dnode_t *dn, dmu_tx_t *tx)
 	dprintf_ds(os->os_dsl_dataset, "obj=%llu txg=%llu\n",
 	    dn->dn_object, txg);
 
-	if (dn->dn_free_txg > 0 && dn->dn_free_txg <= txg) {
-		list_insert_tail(&os->os_free_dnodes[txg&TXG_MASK], dn);
-	} else {
-		list_insert_tail(&os->os_dirty_dnodes[txg&TXG_MASK], dn);
-	}
+	multilist_insert(&os->os_dirty_dnodes[txg&TXG_MASK], dn);
```

Comment:
[cstyle] It would be nice to add the white space back around the `&` in `[txg&TXG_MASK]`.